src/nft/components/sell/select/SelectPage.tsx

```diff
@@ -1,3 +1,380 @@
+  const collectionFilters = useWalletCollections((state) => state.collectionFilters)
+
+  const { data: ownerCollections } = useQuery(
+    ['ownerCollections', address],
+    () => OSCollectionsFetcher({ params: { asset_owner: address, offset: '0', limit: '300' } }),
+    {
+      refetchOnWindowFocus: false,
+    }
+  )
+
+  const ownerCollectionsAddresses = useMemo(() => ownerCollections?.map(({ address }) => address), [ownerCollections])
+  const { data: collectionStats } = useQuery(
+    ['ownerCollectionStats', ownerCollectionsAddresses],
+    () => fetchMultipleCollectionStats({ addresses: ownerCollectionsAddresses ?? [] }),
+    {
+      refetchOnWindowFocus: false,
+    }
+  )
+
+  const {
+    data: ownerAssetsData,
+    fetchNextPage,
+    hasNextPage,
+    isSuccess,
+  } = useInfiniteQuery(
+    ['ownerAssets', address, collectionFilters],
+    async ({ pageParam = 0 }) => {
+      return await fetchWalletAssets({
+        ownerAddress: address ?? '',
+        collectionAddresses: collectionFilters,
+        pageParam,
+      })
+    },
+    {
+      getNextPageParam: (lastPage, pages) => {
+        return lastPage?.flat().length === 25 ? pages.length : null
+      },
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  )
+
+  const ownerAssets = useMemo(() => (isSuccess ? ownerAssetsData?.pages.flat() : null), [isSuccess, ownerAssetsData])
+
+  const walletAssets = useWalletCollections((state) => state.walletAssets)
+  const setWalletAssets = useWalletCollections((state) => state.setWalletAssets)
+  const displayAssets = useWalletCollections((state) => state.displayAssets)
+  const setDisplayAssets = useWalletCollections((state) => state.setDisplayAssets)
+  const setWalletCollections = useWalletCollections((state) => state.setWalletCollections)
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const reset = useSellAsset((state) => state.reset)
+  const setSellPageState = useSellPageState((state) => state.setSellPageState)
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    setWalletAssets(ownerAssets?.flat() ?? [])
+  }, [ownerAssets, setWalletAssets])
+
+  useEffect(() => {
+    ownerCollections && setWalletCollections(ownerCollections)
+  }, [ownerCollections, setWalletCollections])
+
+  const listFilter = useWalletCollections((state) => state.listFilter)
+
+  useEffect(() => {
+    if (searchText) {
+      const filtered = walletAssets.filter((asset) => asset.name?.toLowerCase().includes(searchText.toLowerCase()))
+      setDisplayAssets(filtered, listFilter)
+    } else {
+      setDisplayAssets(walletAssets, listFilter)
+    }
+  }, [searchText, walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    setDisplayAssets(walletAssets, listFilter)
+  }, [walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    if (ownerCollections?.length && collectionStats?.length) {
+      const ownerCollectionsCopy = [...ownerCollections]
```

Comment:
i don't think this will truly copy the collections as it'll still be a reference. if you want to create new things i'd suggest this:

```
ownerCollectionsCopy = ownerCollection.map(collection => {
  return {
    ...collection,
    floorPrice: 1000,
  }
})
```

Reply:
This does make a deep copy. Evidenced here: https://stackoverflow.com/questions/7486085/copy-array-by-value

> The [ ...myArray ], myArray.splice(0), myArray.slice(), and myArray.concat() techniques can be used to deep copy arrays with literal values

This also handles the functionality for which a copy is needed as evidenced on interface-nft. All of the functionality for this code has already been tested in interface-nft

Reply:
oh wild, didn't know that

Reply:
<img width="669" alt="Screen Shot 2022-09-14 at 12 09 34 PM" src="https://user-images.githubusercontent.com/4899429/190206444-3d21e7a1-1e55-4e31-9ef4-4a682df76434.png">

Reply:
<img width="234" alt="Screen Shot 2022-09-14 at 12 10 52 PM" src="https://user-images.githubusercontent.com/4899429/190206717-81c64425-4946-4603-b8ef-0c5cad109bea.png">

Reply:
i think it works regardless so just a nit but wanted to point it out

Reply:
sorry about my previous comment (deleted it) - read vig's code as charlie's code lmao. Vig is right by doing it `[...obj]` you are copying the object by value but the actual properties are passed by call-by-reference in JS (its dumb). Its weird so the right way to take a copy of an array is vig's method